### DISCUSSION TITLE
extra key exchange no work.

### DISCUSSION THREAD

**Discussion by wangxin688** [scrapli/scrapligo#203] [Q&A] [answered]
```go
p, err := platform.NewPlatform(
		driverPlatform,
		connectionInfo.ManagementIp,
		options.WithAuthNoStrictKey(),
		options.WithAuthPassword(connectionInfo.Password),
		options.WithAuthUsername(connectionInfo.Username),
		options.WithPort(connectionInfo.Port),
		options.WithTimeoutOps(timeout),
		options.WithStandardTransportExtraCiphers([]string{
			"aes128-ctr", "aes192-ctr",
			"aes256-ctr", "aes128-gcm@openssh.com",
			"aes256-gcm@openssh.com", "chacha20-poly1305@openssh.com",
			"arcfour256", "arcfour128", "aes128-cbc",
			"3des-cbc", "aes192-cbc", "aes256-cbc",
		}),
		options.WithStandardTransportExtraKexs([]string{
			"diffie-hellman-group14-sha1",
			"diffie-hellman-group-exchange-sha1"}),
	)
```
I got some old devices which need support `diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha1` key exchange, seems it's not work when I add options to `NewPlatform`.
error message:
`"error": "[sshConnection]: failed to open ssh connection: errConnectionError: encountered error output during in channel ssh authentication, error: 'no matching key exchange found for host, their offer: Their offer: diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha1'"}`

**Comment by wangxin688** [comment]
or if there's a option to make `ssh.ClientConfig` with `ssh.InsecureIgnoreHostKey()`, so I can ignore the key exchange. although I known it's not recommend

**Reply by wangxin688** [reply]
> or if there's a option to make `ssh.ClientConfig` with `ssh.InsecureIgnoreHostKey()`, so I can ignore the key exchange. although I known it's not recommend

seems `WithAuthNoStrictKey` is ignore key exchange not match.

**Comment by carlmontanari** [comment, marked as answer]
It's a little confusing, but I think you re looking for `WithSystemTransportOptions` -- tl;dr "system" == /bin/ssh, "standard" == crypto/ssh. the "system" was original name in python version, then when we added the crypto/ssh thing we called it "standard" because standard library stuff.

example in the tests [here](https://github.com/scrapli/scrapligo/blob/eef59b6bb4d30dc0e6b8db8773d406f9b4f1dbb7/driver/network/driver_test.go#L265) that should be relevant

**Reply by wangxin688** [reply]
that's great. thanks for elaborating, it's help a lot, I will take a look into more unit test cases. BTW, any plans to and some FAQs to readme.md? I also encounter with ` failed to open ssh connection: read /dev/ptmx: input/output error` randomly, but after I checked with closed issues, it was solved. If someone get the same and frequently problems it can be resovled quickly

**Reply by carlmontanari** [reply]
to be honest, no, no plans. I know thats not a "good" answer, but its the truth :) 

folks like you opening discussions like this is obviously great and helpful, but thats kind of it for now. I wrote a ton of docs for (python) scrapli before and they may be worth a read for scrapligo users as well as scrapligo is really a pretty faithful port of the python version, but thats pretty much it for docs at the moment.

**Reply by wangxin688** [reply]
got it, anyway, thanks for update, sure I will read the python version docs
